src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -140,53 +134,48 @@ public class Worker {
+          "Graceful Shutdown - The current worker will not be registered again and should be shutdown gracefully!");
+      pipeline.stopMatchingOperations();
+      int scanRate = 30; // check every 30 seconds
+      int timeWaited = 0;
+      int timeOut = configs.getWorker().getGracefulShutdownSeconds();
 
-    try {
-      while (!pipeline.isEmpty() && timeWaited < timeOut) {
-        SECONDS.sleep(scanRate);
-        timeWaited += scanRate;
-        log.log(INFO, String.format("Pipeline is still not empty after %d seconds.", timeWaited));
-      }
-    } catch (InterruptedException e) {
-      log.log(Level.SEVERE, "The worker gracefully shutdown is interrupted: " + e.getMessage());
-    } finally {
-      // make a grpc call to disable scale protection
-      String clusterEndpoint = configs.getServer().getAdmin().getClusterEndpoint();
-      log.log(
-          INFO,
-          String.format(
-              "It took the worker %d seconds to %s",
-              timeWaited,
-              pipeline.isEmpty() ? "finish all actions" : "but still cannot finish all actions"));
       try {
-        awsAdmin.disableHostScaleInProtection(clusterEndpoint, configs.getWorker().getPublicName());
-      } catch (Exception e) {
-        log.log(
-            SEVERE,
+        if (pipeline.isEmpty()) {
+          System.err.println("Graceful Shutdown - no work in the pipeline.");
+        } else {
+          System.err.println(
+              String.format("Graceful Shutdown - waiting for executions to finish."));
+        }
+        while (!pipeline.isEmpty() && timeWaited < timeOut) {
+          SECONDS.sleep(scanRate);
+          timeWaited += scanRate;
+          System.err.println(
+              String.format(
+                  "Graceful Shutdown - Pipeline is still not empty after %d seconds.", timeWaited));
+        }
+      } catch (InterruptedException e) {
+        System.err.println(
+            "Graceful Shutdown - The worker gracefully shutdown is interrupted: " + e.getMessage());
+      } finally {
+        System.err.println(
             String.format(
-                "gRPC call to AdminService to disable scale in protection failed with exception: %s and stacktrace %s",
-                e.getMessage(), Arrays.toString(e.getStackTrace())));
-        // Gracefully shutdown cannot be performed successfully because of error in
-        // AdminService side. Under this scenario, the worker has to be added back to the worker
-        // pool.
-        inGracefulShutdown = false;
+                "Graceful Shutdown - It took the worker %d seconds to %s",
+                timeWaited,
+                pipeline.isEmpty()
+                    ? "finish all actions"
+                    : "gracefully shutdown but still cannot finish all actions"));
       }
+    } else {
+      System.err.println(
+          String.format(
+              "Graceful Shutdown is not enabled. Worker is shutting down without finishing executions in progress."));
```

Comment:
i would prefer to do this branch first since the block is shorter. but no big deal. 
```
if (configs.getWorker().getGracefulShutdownSeconds() ==0) {
  System.err.println(...);
} else {
  ...
}
```